x/stake/handler.go

```diff
@@ -104,6 +104,7 @@ func handleMsgCreateValidator(ctx sdk.Context, msg types.MsgCreateValidator, k k
 	}
 
 	validator := NewValidator(msg.ValidatorAddr, msg.PubKey, msg.Description)
+	validator.BondHeight = ctx.BlockHeight()
```

Comment:
Is this the best place for the change given that a validator may still not enter the bonded set here? Just thinking about semantics...

Reply:
This doesn't make sense - bond height is supposed to be set when the validator is bonded, not when it is created. This code is erroneous if you are creating a validator with "tiny" stake for chain where the validator group is already full

Reply:
yeah. I think we can close this PR and another one be opened that addresses the deeper problem.